File: EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/VisitorEntryLog.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class VisitorEntryLog : BaseAuditableEntity
{
    public long SchoolId { get; set; }
    public string VisitorFullName { get; set; } = string.Empty;
    public string NationalIdOrPassport { get; set; } = string.Empty;
    public string VisitPurpose { get; set; } = string.Empty;
    public long? HostEmployeeId { get; set; }
    public DateTime CheckInTime { get; set; } = DateTime.UtcNow;
    public DateTime? CheckOutTime { get; set; }
    public string? VisitorBadgeNumber { get; set; }
    public int Status { get; set; } // 1=InCampus, 2=Departed, 3=Flagged
    public string? VisitorPhoneNumber { get; set; }
    public string? VisitorOrganization { get; set; }
    public string? SecurityGateNumber { get; set; }
    public long? SecurityOfficerEmployeeId { get; set; }

    // Navigation Properties
    public virtual School? School { get; set; }
    public virtual Employee? HostEmployee { get; set; }
    public virtual Employee? SecurityOfficerEmployee { get; set; }
}

File: src/app/modules/m1-school-office/data-access/models/visitor-entry-log.ts
import type { VisitStatus } from './common';

export interface VisitorEntryLog {
  id: number;
  schoolId: number;
  visitorFullName: string;
  nationalIdOrPassport: string;
  visitPurpose: string;
  hostEmployeeId: number | null;
  checkInTime: string;
  checkOutTime: string | null;
  visitorBadgeNumber: string | null;
  status: VisitStatus;
  visitorPhoneNumber: string | null;
  visitorOrganization: string | null;
  securityGateNumber: string | null;
  securityOfficerEmployeeId: number | null;
}

export type CreateVisitorEntryLogDto = Omit<VisitorEntryLog, 'id' | 'status'>;

export type UpdateVisitorEntryLogDto = Omit<VisitorEntryLog, 'schoolId' | 'status'>;
